lib/charms/postgresql_k8s/v1/postgresql.py

```diff
@@ -976,8 +978,9 @@ def set_up_predefined_catalog_roles_function(self) -> None:
 BEGIN
 	database := (SELECT current_database());
 	current_session_user := (SELECT session_user);
-    owner_user := database || '_owner';
-    admin_user := database || '_admin';
+    owner_user := quote_ident(database || '_owner');
+    admin_user := quote_ident(database || '_admin');
+    database := quote_ident(database);
```

Comment:
When a database name contains a hyphen (like in `test-database`), it's needed to quote those identifiers, otherwise the SQL statements using them will fail.